module-api/src/main/java/kitchenpos/menu/application/MenuService.java

```diff
@@ -24,19 +24,25 @@ public MenuService(MenuRepository menuRepository, MenuGroupService menuGroupServ
 
     public MenuResponse save(MenuRequest request) {
         MenuGroup menuGroup = menuGroupService.findById(request.getMenuGroupId());
-        Menu menu = Menu.of(request.getName(), request.getPrice(), menuGroup);
+        final Menu savedMenu = menuRepository.save(Menu.of(request.getName(), request.getPrice(), menuGroup));
 
         Map<Long, Product> productById = getProducts(request);
-        request.getMenuProductRequests()
-            .forEach(menuProductRequest -> menu.addMenuProduct(
-                productById.get(menuProductRequest.getProductId()),
-                menuProductRequest.getQuantity()
-            ));
 
-        return MenuResponse.from(menuRepository.save(menu));
+        if (Objects.nonNull(productById)) {
+            request.getMenuProductRequests()
+                .forEach(menuProductRequest -> savedMenu.addMenuProduct(
+                    productById.get(menuProductRequest.getProductId()),
+                    menuProductRequest.getQuantity()
+                ));
+        }
+
+        return MenuResponse.from(menuRepository.save(savedMenu));
     }
 
     private Map<Long, Product> getProducts(MenuRequest request) {
+        if (Objects.isNull(request.getMenuProductRequests())) {
+            return null;
```

Comment:
```suggestion
            return Collections.emptyMap();
```

컬렉션의 경우 null을 사용하기 보다 빈 값을 사용해보면 어떨까요?

Reply:
넵, 감사합니다. 이런 것이 있는 줄도 몰랐네요. 😭 
자바를 많이 사용해도 사용하는 것만 사용하다보니, 리뷰어님께서 알려주셔서 배워갑니다.